pkg/providers/tinkerbell/validator.go

```diff
@@ -132,7 +134,28 @@ func (v *Validator) ValidateHardwareConfig(ctx context.Context, hardwareConfigFi
 		return fmt.Errorf("failed validating Secrets in hardware config: %v", err)
 	}
 
-	logger.MarkPass("Hardware Config is valid")
+	if err := v.ValidateBMCSecretCreds(ctx, v.hardwareConfig); err != nil {
+		return err
+	}
+
+	logger.MarkPass("Hardware Config file validated")
+
+	return nil
+}
+
+func (v *Validator) ValidateBMCSecretCreds(ctx context.Context, hc hardware.HardwareConfig) error {
+	for index, bmc := range hc.Bmcs {
+		bmcInfo := hardware.BmcSecretConfig{
+			Host:     bmc.Spec.Host,
+			Username: string(hc.Secrets[index].Data["username"]),
+			Password: string(hc.Secrets[index].Data["password"]),
+			Vendor:   bmc.Spec.Vendor,
+		}
+		if err := v.pbnj.ValidateBMCSecretCreds(ctx, bmcInfo); err != nil {
+			return fmt.Errorf("failed validating Hardware BMC credentials for the node %v, host %v : %v", hc.Hardwares[index].Name, bmcInfo.Host, err)
```

Comment:
nit: use `%s` format specifier for strings
```suggestion
			return fmt.Errorf("failed validating Hardware BMC credentials for the node %s, host %s : %v", hc.Hardwares[index].Name, bmcInfo.Host, err)
```